### DISCUSSION TITLE
Having trouble compiling C code with gcc in Big Sur

### DISCUSSION THREAD

**Discussion by johncoxon** [Homebrew/discussions#2037] [Everyday usage]
Hi all,

I'm trying to compile some C code and getting the error `ld: library not found for -lc`. I had been having problems with `gfortran` and found [this discussion](https://github.com/Homebrew/discussions/discussions/303), and inserting `export SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk` into my `zshrc` fixed my `gfortran` issues but not the C compilation issues.

Trying to reinstall with `brew reinstall -s gcc` did not fix the issue for the C code, and for `gfortran` every time I try to compile everything it prints `zsh killed gfortran` and failed so I've fallen back to the bottled version and the `SDKROOT` in my `zshrc`.

Can anyone give me any pointers? The C code I'm trying to compile can be found [here](https://spdf.gsfc.nasa.gov/pub/software/cdf/dist/cdf38_0/macosx/src_distribution/cdf38_0-dist-all.tar.gz).

**Comment by SMillerDev** [comment]
Try ensuring that your gcc is up to date and make sure `brew doctor` doesn't give any issues.

**Comment by johncoxon** [comment]
The output of `brew doctor` was as follows:

```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /opt/homebrew/Caskroom/miniforge/base/bin/krb5-config
  /opt/homebrew/Caskroom/miniforge/base/bin/freetype-config
  /opt/homebrew/Caskroom/miniforge/base/bin/libpng16-config
  /opt/homebrew/Caskroom/miniforge/base/bin/libpng-config
  /opt/homebrew/Caskroom/miniforge/base/bin/python3.9-config
  /opt/homebrew/Caskroom/miniforge/base/bin/python3-config
  /opt/homebrew/Caskroom/miniforge/base/bin/curl-config
  /opt/homebrew/Caskroom/miniforge/base/bin/ncursesw6-config
  /opt/homebrew/Caskroom/miniforge/base/bin/ncurses6-config
  /opt/homebrew/Caskroom/miniforge/base/bin/nc-config

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /opt/homebrew/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/opt/homebrew/sbin:$PATH"' >> ~/.zshrc
```

`/opt/homebrew/sbin` just has the following files in it:
```
unbound			unbound-checkconf	unbound-control-setup
unbound-anchor		unbound-control		unbound-host
```

I believe gcc is up-to-date – running `brew upgrade gcc` gives `Warning: gcc 11.2.0 already installed`.

**Reply by SMillerDev** [reply]
I see multiple references to `/Developer/SDKs/MacOSX10.5.sdk` in the Makefile. You can try replacing those with `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk` and see if it helps, but I really don't think this is an issue with anything Homebrew does.

**Reply by johncoxon** [reply]
That caused me to have a poke in the makefile and I noticed that one of the compilers used is clang and that gives the same error, so yes, I think it's an issue in the makefile and not in gcc. Thank you!
